crates/sui-indexer/src/handlers/committer.rs

```diff
@@ -43,7 +44,17 @@ pub async fn start_tx_checkpoint_commit_task<S>(
 
     let mut stream = mysten_metrics::metered_channel::ReceiverStream::new(tx_indexing_receiver)
         .ready_chunks(checkpoint_commit_batch_size);
+
     let mut object_snapshot_backfill_mode = true;
+    let latest_object_snapshot_seq = state
+        .get_latest_object_snapshot_checkpoint_sequence_number()
+        .await?;
+    let latest_cp_seq = state.get_latest_checkpoint_sequence_number().await?;
+    if latest_object_snapshot_seq != latest_cp_seq {
```

Comment:
But wouldn't this always be false?

Reply:
during backfill with the snapshot backfill mode, latest_object_snapshot_seq == latest_cp_seq, b/c all these tables are ingested together with watermark in checkpoints table.

Reply:
but if for some reason, like this time it happened, we restarted indexer when objects_snapshot is behind already, it will try to do backfill mode, which caused the data corruption / missing data.